scripts/deploy.sh

```diff
@@ -90,9 +92,9 @@ if [[ "${DEPLOY_AIO}" == "yes" ]]; then
       echo "osd_directory: true" | tee -a $RPCD_VARS
       echo "osd_directories:" | tee -a $RPCD_VARS
       echo "  - /var/lib/ceph/osd/mydir1" | tee -a $RPCD_VARS
-      sed -i "s/glance_default_store:.*/glance_default_store: rbd/" /etc/openstack_deploy/user_variables.yml
-      echo "nova_libvirt_images_rbd_pool: vms" | tee -a /etc/openstack_deploy/user_variables.yml
-      echo "cinder_ceph_client_uuid:"  | tee -a /etc/openstack_deploy/user_secrets.yml
+      sed -i "s/glance_default_store:.*/glance_default_store: rbd/" /etc/openstack_deploy/user_osa_variables_defaults.yml
+      echo "nova_libvirt_images_rbd_pool: vms" | tee -a /etc/openstack_deploy/user_rpco_variables_defaults.yml
+      echo "cinder_ceph_client_uuid:"  | tee -a /etc/openstack_deploy/user_rpco_variables_defaults.yml
```

Comment:
cinder_ceph_client_uuid was being appended to user_secrets and is now being appended to defaults. I think this line should be removed as cinder_ceph_client_uuid is in OSA/user_secrets already. This is probably causing the problem as the cinder_ceph_client_uuid is specified with a null value (to be replaced by the pwgen script) but pwgen is not run against the defaults file. The defaults file is then specified after the secrets file by openstack-ansible, so null from user_rpco_variables_defaults takes precedent over the generated secret from user_osa_secrets_defaults.yml